### PR TITLE
[CSL-1453] Remove extraneous dependencies

### PR DESCRIPTION
**Pull request by rockbmb**
@neongreen I added this PR's commit to `extra-deps` in `cardano-sl` and it builds OK.

### CODE REVIEW

File: src/Serokell/Data/Variant/Serialization.hs [hunk 1]
```diff
@@ -87,6 +92,39 @@ instance Aeson.FromJSON Variant where
...
+genericPutVector :: (Cereal.Serialize a, VG.Vector v a) => Cereal.Putter (v a)
+{-# INLINE genericPutVector #-}
+genericPutVector = genericPutVectorWith Cereal.put
+
+-- | Read a 'Data.Vector.Generic.Vector'.
+genericGetVector :: (Cereal.Serialize a, VG.Vector v a) => Cereal.Get (v a)
+{-# INLINE genericGetVector #-}
+genericGetVector = genericGetVectorWith Cereal.get
+
+instance (Cereal.Serialize a) => Cereal.Serialize (V.Vector a) where
```
Comment (ChShersh) [inline src/Serokell/Data/Variant/Serialization.hs]: Do we still need `cereal` and `binary`? We are using `cbor` and don't use other serialization formats. Maybe, there's not point in depending on `cereal` and write all these helpers. This is meta-question to all interested in future of `serokell-util`.

File: src/Serokell/Util/Base64.hs [hunk 2]
```diff
@@ -92,3 +91,11 @@ instance FromJSON JsonByteStringDeprecated where
     parseJSON =
         parseJSON >=>
         either (fail . T.unpack) (pure . JsonByteStringDeprecated) . decodeUrl
+
+----------------------------------------------------------------------------
+-- Decoding helpers
+----------------------------------------------------------------------------
+
+mapLeft :: (a -> c) -> Either a b -> Either c b
```
Comment (ChShersh) [inline src/Serokell/Util/Base64.hs]: `mapLeft` is just `first` from `Data.Bifunctor`, no need in extra function.

File: src/Serokell/Data/Variant/Serialization.hs (hunk 1 above)
Comment (neongreen) [inline src/Serokell/Data/Variant/Serialization.hs]: I think that it's better to import orphan instances from their respective packages than to put them into serokell-util.

Comment (ChShersh) [inline src/Serokell/Data/Variant/Serialization.hs]: Yes, in perfect world there's no `serokell-util`. I just saying that maybe we should remove `cereal` and `binary` from dependencies here.

**Comment by rockbmb** [comment]
@neongreen I don't understand what you mean here. If it's better to import the orpgan instances then I shouldn't have removed the packages in the first place, correct?

File: src/Serokell/Util/Base64.hs (hunk 2 above)
Comment (dcoutts) [inline src/Serokell/Util/Base64.hs]: Yes, if we can drop cereal and binary deps here that'd be good. I don't think we have other packages using cereal and binary much (and if they still do, that's a good target for pruning too).

**Comment by dcoutts** [comment]
Yes, if we can drop cereal and binary deps here that'd be good. I don't think we have other packages using cereal and binary much (and if they still do, that's a good target for pruning too).

**Comment by neongreen** [comment]
@rockbmb 

> If it's better to import the orpgan instances then I shouldn't have removed the packages in the first place

No, you should've. My point was that if we need the orphans, we can import them in CSL, not in serokell-util.

File: src/Serokell/Util/Base64.hs (hunk 2 above)
Comment (rockbmb) [inline src/Serokell/Util/Base64.hs]: Good find, didn't notice that one.

**Pull request merged**